libs/parallelism/algorithms/include/hpx/parallel/util/detail/sender_util.hpp

```diff
@@ -0,0 +1,125 @@
+#include <hpx/execution/algorithms/transform.hpp>
+#include <hpx/execution/traits/is_execution_policy.hpp>
+#include <hpx/execution_base/sender.hpp>
+#include <hpx/executors/execution_policy.hpp>
+
+#include <type_traits>
+#include <utility>
+
+namespace hpx { namespace detail {
+    // This is a lighter-weight alternative to bind for use in parallel
+    // algorithm overloads, where one needs to bind an execution policy to an
+    // algorithm for use in execution::transform. Typically used together with
+    // transform_with_bound_algorithm.
+    template <typename Tag, typename ExPolicy>
+    struct bound_algorithm
+    {
+        std::decay_t<ExPolicy> policy;
+
+        template <typename T1, typename... Ts>
+        auto operator()(T1&& t1, Ts&&... ts) -> decltype(Tag{}(
+            std::move(policy), std::forward<T1>(t1), std::forward<Ts>(ts)...))
+        {
+            return Tag{}(std::move(policy), std::forward<T1>(t1),
+                std::forward<Ts>(ts)...);
+        }
+    };
+
+    // Detects if the given type is a bound_algorithm.
+    template <typename Bound>
+    struct is_bound_algorithm : std::false_type
+    {
+    };
+
+    template <typename Tag, typename ExPolicy>
+    struct is_bound_algorithm<bound_algorithm<Tag, ExPolicy>> : std::true_type
+    {
+    };
+
+    // Helper function for use in creating overloads of parallel algorithms that
+    // take senders. Takes an execution policy, a predecessor sender, and an
+    // "algorithm" (i.e. a tag) and applies transform with the predecessor
+    // sender and the execution policy bound to the algorithm.
+    template <typename Tag, typename ExPolicy, typename Predecessor>
+    auto transform_with_bound_algorithm(
+        Predecessor&& predecessor, ExPolicy&& policy)
+    {
+        // If the given execution policy can has a task policy, i.e. the
+        // algorithm can return a future, we use the task policy since we can
+        // then directly return the future as a sender and avoid potential
+        // blocking that may happen internally.
+        if constexpr (hpx::execution::detail::has_async_execution_policy_v<
+                          ExPolicy>)
+        {
+            auto task_policy =
+                std::forward<ExPolicy>(policy)(hpx::execution::task);
+            return hpx::execution::experimental::let_value(
+                std::forward<Predecessor>(predecessor),
+                bound_algorithm<Tag, decltype(task_policy)>{
+                    std::move(task_policy)});
+        }
+        // If the policy does not have a task policy, the algorithm can only be
+        // called synchronously. In this case we only use transform to chain the
+        // algorithm after the predecessor sender.
+        else
+        {
+            return hpx::execution::experimental::transform(
+                std::forward<Predecessor>(predecessor),
+                bound_algorithm<Tag, ExPolicy>{std::forward<ExPolicy>(policy)});
+        }
```

Comment:
I think that we should create the task policies where missing. `unseq` and `par_unseq` simply map onto `par` and `seq` anyways... I have created a ticket for us not to forget: https://github.com/STEllAR-GROUP/hpx/issues/5492.

Reply:
Thanks for opening the issue! I forgot to mention that I went the shorter route of removing the task overloads for those policies, but you noticed anyway ;)